### PR TITLE
Master beercrud

### LINKED ISSUE #3: Admin: Beers
Create CRUD-functionality for beers

- [x] List in table, with pagination
- [x] Add new, top of table
- [x] Validation on input for adding new records
- [x] Edit (table functionality available?)
- [x] Delete (table functionality available?)

### PR DESCRIPTION
**Pull request by askaret**
closes #3

### CODE REVIEW

File: BlazorTest/Pages/admin/beer/Beer.razor [hunk 1]
```diff
@@ -3,269 +3,196 @@
...
+	private bool TableFilterFunction(BeerModel beer) => TableFilter(beer, tableSearchString);
+
+	private async void DeleteBeer(BeerModel beer)
+	{
+		var parameters = new DialogParameters();
+		parameters.Add("ContentText", $"Confirm deletion of \"{beer.Name}\"");
+		parameters.Add("ButtonText", "DELETE");
+		parameters.Add("Color", Color.Error);
+
+		var options = new DialogOptions() { CloseButton = true, MaxWidth = MaxWidth.ExtraSmall };
```
Comment (lowren) [inline BlazorTest/Pages/admin/beer/Beer.razor:129]: Is DialogOptions in use?

File: BlazorTest/Pages/admin/beer/Beer.razor [hunk 2]
```diff
@@ -3,269 +3,196 @@
...
+			<MudTd DataLabel="URL">
+				@{
+					if (!string.IsNullOrEmpty(context?.RateBeerLink))
+					{
+					<a href="@context.RateBeerLink" target="_blank"><MudIcon Icon="@Icons.Material.Outlined.Link" Color="Color.Primary" /></a>
+					}
+			}
+		</MudTd>
+		<MudTd>
+			<MudButton Variant="Variant.Outlined" Color="Color.Primary" Size="Size.Small" OnClick="@(() => EditBeer(context))">EDIT</MudButton>
```
Comment (lowren) [inline BlazorTest/Pages/admin/beer/Beer.razor:97]: I dont have setup to debug the changes, but would this look smoother with MudIconButtons?

File: BlazorTest/Pages/admin/beer/Beer.razor [hunk 3]
```diff
@@ -3,269 +3,196 @@
...
+			<MudTd DataLabel="Name">@context.Name</MudTd>
+			<MudTd DataLabel="Brewery">@(context.BreweryModel == null ? "" : context.BreweryModel.Name)</MudTd>
+			<MudTd DataLabel="ABV %">@context.ABV</MudTd>
+			<MudTd DataLabel="Style">@(context.BeerStyleModel == null ? "" : context.BeerStyleModel.Name)</MudTd>
+			<MudTd DataLabel="Class">@(context.BeerclassModel == null ? "" : context.BeerclassModel.Name)</MudTd>
+			<MudTd DataLabel="URL">
+				@{
+					if (!string.IsNullOrEmpty(context?.RateBeerLink))
+					{
+					<a href="@context.RateBeerLink" target="_blank"><MudIcon Icon="@Icons.Material.Outlined.Link" Color="Color.Primary" /></a>
```
Comment (lowren) [inline BlazorTest/Pages/admin/beer/Beer.razor:92]: Is there a difference between this and MudIconButton?

File: BlazorTest/Pages/admin/beer/Beer.razor [hunk 4]
```diff
@@ -3,269 +3,196 @@
...
+	private async Task InsertBeer()
+	{
+		success = true;
+		var added = new BeerModel()
+			{
+				Name = newBeer.Name,
+				ABV = newBeer.ABV,
+				RateBeerLink = newBeer.RateBeerLink,
+				BeerClassId = newBeer.BeerClassId,
+				BeerStyleId = newBeer.BeerStyleId,
```
Comment (lowren) [inline BlazorTest/Pages/admin/beer/Beer.razor:187]: Is beer style ID, beer class ID, brewery ID needed in BeerViewModel? Their model classes are expected to not be null below.

Comment (askaret) [reply BlazorTest/Pages/admin/beer/Beer.razor:187]: It is used when editing a beer, see the Beer\Edit.blazor page for deets

File: BlazorTest/Pages/admin/beer/Beer.razor (hunk 1 above)
Comment (askaret) [reply BlazorTest/Pages/admin/beer/Beer.razor:129]: No, removed

File: BlazorTest/Pages/admin/beer/Beer.razor (hunk 3 above)
Comment (askaret) [reply BlazorTest/Pages/admin/beer/Beer.razor:92]: Yup. MudIcon is just an icon and in this case I've wrapped it with an <a>....</a> tag to make it clickable. A MudIconButton would need a function to do the same

File: BlazorTest/Pages/admin/beer/Beer.razor (hunk 4 above)
Comment (lowren) [reply BlazorTest/Pages/admin/beer/Beer.razor:187]: Agreed, disregard original comment

File: BlazorTest/Pages/admin/beer/Beer.razor (hunk 2 above)
Comment (askaret) [reply BlazorTest/Pages/admin/beer/Beer.razor]: Yeah, it did. So I changed them :)

**Pull request merged**